### ISSUE TITLE
TinyMCE strips WAI-ARIA attributes

### ISSUE THREAD

**Issue by paulschreiber** [tinymce/tinymce#3402]
WAI-ARIA attributes are stripped by TinyMCE. Attributes that should be allowed:
role
aria-hidden
aria-checked
aria-selected
aria-pressed
aria-label
aria-labelledby
aria-disabled
aria-grabbed
aria-controls
aria-describedby
aria-required
tabindex

Workaround for now (in WordPress):

```
add_filter( 'tiny_mce_before_init', 'tinymce_allow_tags' );

public static function tinymce_allow_tags( $options ) {
	if ( ! isset( $options['extended_valid_elements'] ) ) {
		$options['extended_valid_elements'] = '';
	}

	$options['valid_elements'] .= ',@[role|aria-hidden|aria-checked|aria-selected|aria-pressed|aria-label|aria-labelledby|aria-disabled|aria-grabbed|aria-controls|aria-describedby|aria-required|tabindex]';

	return $options;
}
```

**Comment by ShirlGirl** [comment]
I need to be able to include Aria labels on my sites. I like the idea of a work around; however, I am a total neophyte and do not know where to place the add filter. Is it to go into the TinyMCE plug-in functions.php file or into the theme's?

**Comment by paulschreiber** [comment]
Most of these are fixed in 4.6.3. `role` and `tabindex` are still stripped.

**Comment by lostkeys** [comment]
This old issue seems to be fixed somewhere along the way. See http://fiddle.tinymce.com/Nhhaab. Please reopen the issue if this isn't the case.
